### PR TITLE
Bls implementation using Apache Milagro

### PR DESCRIPTION
**Pull request by mpetrunic**
I didn't get answer from herumi about mcl-wasm (eth2 requires changing quite a lot of stuff in mcl-wasm library because of serialization), so this is implementation using apache milagro. 
**Passes all spec tests.**
In future we can see and migrate to mcl-wasm depending on beanchmarks.

I also modified apache milagro libary (added typescript type difinitions and module exports so it can work with Node/Typescript). You would probably wan't to fork it : https://github.com/mpetrunic/incubator-milagro-crypto

This just implements method tested in spec, I will add rest of methods, travis config, migration to published eth2.0 test helpers in subsequent PRs.

Cheers!

### CODE REVIEW

**Comment by GregTheGreek** [comment]
You should make an upstream PR for https://github.com/mpetrunic/incubator-milagro-crypto imo I think they would accept it :)

**Review by GregTheGreek — APPROVED** [review]

**Comment by mpetrunic** [comment]
Yeah, sure do but I'm still missing tons of types and currently it will break their standalone web usage(I can fix it but it's a lot of repetative work)

**Comment by GregTheGreek** [comment]
Oh I see, also I believe you should have access to fork into chainsafe if you want

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import {BLSDomain, BLSPrivKey, BLSPubkey, BLSSignature, bytes32} from "./types";
+import {Keypair} from "./keypair";
+import {PrivateKey} from "./privateKey";
+import {G2point} from "./helpers/g2point";
+import {PublicKey} from "./publicKey";
+import {G1point} from "./helpers/g1point";
+
+export default class BLS {
+
+  public static generatePublicKey(privateKey: BLSPrivKey): BLSPubkey {
```
Comment (wemeetagain) [inline src/index.ts:10]: I think we can just export these functions directly instead of on a class

File: src/types.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export type bytes8 = Buffer;
+export type bytes32 = Buffer;
+export type bytes48 = Buffer;
+export type bytes96 = Buffer;
+
+export const bytes8 = "bytes8";
```
Comment (wemeetagain) [inline src/types.ts:6]: I don't think we need these consts here.

**Review by wemeetagain — COMMENTED** [review]
Late review

File: src/types.ts (hunk 2 above)
Comment (mpetrunic) [reply src/types.ts:6]: yeah probably, I will make some refactoring to name all the types, this was just assembled to see if I can pass spec tests. Now it's trivial to finish rest of the functions and clean up

**Pull request merged**